server/src/main/java/com/linecorp/centraldogma/server/internal/api/ContentServiceV1.java

```diff
@@ -436,8 +438,24 @@ public static void checkMirrorLocalRepo(String repoName, Iterable<Change<?>> cha
                                return null;
                            }).filter(Objects::nonNull).findFirst();
             if (notAllowedLocalRepo.isPresent()) {
-                throw new IllegalArgumentException("invalid " + MIRROR_LOCAL_REPO + ": " +
-                                                   notAllowedLocalRepo.get());
+                throw new InvalidPushException("invalid " + MIRROR_LOCAL_REPO + ": " +
+                                               notAllowedLocalRepo.get());
+            }
+        }
+    }
+
+    /**
+     * Checks if the commit is for creating a file and raises a {@link InvalidPushException} if the
+     * given {@code repoName} field is one of {@code meta} and {@code dogma} which are internal repositories.
+     */
+    public static void checkPushLocalRepo(String repoName, Iterable<Change<?>> changes) {
+        if (Project.isReservedRepoName(repoName)) {
```

Comment:
Right, it would be better to refuse all changes to the `dogma` repository.

While applying, I merged `checkMirrorLocalRepo` and `checkPushLocalRepo` to `checkPush`. Is it fine for you? Would it be breaking changes?

Reply:
> While applying, I merged checkMirrorLocalRepo and checkPushLocalRepo to checkPush. Is it fine for you? Would it be breaking changes?

I think it's okay because the client will get a CentralDogmaException anyway.
So, let's merge them. 😉